locust_plugins/embedded_resource_manager.py

```diff
@@ -0,0 +1,179 @@
+
+        self.resource_paths = [
+            '//link[@rel="stylesheet"]/@href',
+            '//link[@rel="Stylesheet"]/@href',
+            '//link[@rel="STYLESHEET"]/@href',
+            "//script/@src",
+            "//img/@src",
+            "//source/@src",
+            "//embed/@src",
+            "//body/@background",
+            '//input[@type="image"]/@src',
+            '//input[@type="IMAGE"]/@src',
+            '//input[@type="Image"]/@src',
+            "//object/@data",
+            "//frame/@src",
+            "//iframe/@src",
+        ]
+
+        self.client = user.client
+        self.client.request = self._request(self.client.request)
+        self.host = user.host
+
+    def get_embedded_resources(self, response_content, resource_filter):
+        """
+        returns a list of embedded resources in response_content
+        provide a regex filter to limit what resources are returned
+        """
+        resources = []
+        # check if defaults have been overridden for this request
+        if self.cache_resource_links and response_content in self.resource_link_cache:
+            resources = self.resource_link_cache[response_content]
+        else:
+            try:
+                tree = html.fromstring(response_content)
+                # check for base tag - otherwise use host for partial urls
+                base_path_links = tree.xpath("//base/@href")
+                base_path = (
+                    base_path_links[0] if len(base_path_links) > 0 else self.host
+                )
+                # build resource list
+                for resource_path in self.resource_paths:
+                    for resource in tree.xpath(resource_path):
+                        if re.search(self.full_url_pattern, resource) is None:
+                            resource = base_path + "/" + resource
+                        if re.search(resource_filter, resource):
+                            resources.append(resource)
+                # add style urls
+                style_tag_texts = tree.xpath("//style/text()")
+                for text in style_tag_texts:
+                    # check for url
+                    url_matches = re.match(self.url_link_pattern, text)
+                    if url_matches is not None:
+                        resource = url_matches[2]
+                        if re.search(self.full_url_pattern, resource) is None:
+                            resource = base_path + "/" + resource
+                        if re.search(resource_filter, resource):
+                            resources.append(resource)
+                if self.cache_resource_links:
+                    self.resource_link_cache[response_content] = resources
+            except etree.ParserError as e:
+                logging.warning(str(e) + " " + str(response_content))
+        return resources
+
+    def _request(self, func):
+        def wrapper(*args, **kwargs):
```

Comment:
why not use named args with default values? all this kwargs stuff is confusing :)

Reply:
like this `def wrapper(*args, include_resources=self.include_resources, ..., **kwargs)`

Reply:
good idea - this looks much cleaner